crates/polars-stream/src/nodes/parquet_source/row_group_data_fetch.rs

```diff
@@ -197,46 +197,32 @@ impl RowGroupDataFetcher {
                             mem_slice,
                         }
                     } else if let Some(columns) = projection.as_ref() {
-                        let ranges = get_row_group_byte_ranges_for_projection(
+                        let mut ranges = get_row_group_byte_ranges_for_projection(
                             &row_group_metadata,
                             columns.as_ref(),
                         )
                         .collect::<Vec<_>>();
 
-                        let bytes = current_byte_source.get_ranges(ranges.as_ref()).await?;
+                        let n_ranges = ranges.len();
 
-                        assert_eq!(bytes.len(), ranges.len());
+                        let bytes_map = current_byte_source.get_ranges(&mut ranges).await?;
 
-                        let mut bytes_map = PlHashMap::with_capacity(ranges.len());
-
-                        for (range, bytes) in ranges.iter().zip(bytes) {
-                            memory_prefetch_func(bytes.as_ref());
-                            let v = bytes_map.insert(range.start, bytes);
-                            debug_assert!(v.is_none(), "duplicate range start {}", range.start);
-                        }
+                        assert_eq!(bytes_map.len(), n_ranges);
 
                         FetchedBytes::BytesMap(bytes_map)
                     } else {
-                        // We have a dedicated code-path for a full projection that performs a
-                        // single range request for the entire row group. During testing this
-                        // provided much higher throughput from cloud than making multiple range
-                        // request with `get_ranges()`.
```

Comment:
We had this previously, which was fast if the file contained many nicely sized row groups, but not for the 12,000 column single row group file.

We now just use `get_ranges()`, which handles the download optimization for us